include/rmm/logger.hpp

```diff
@@ -107,7 +107,10 @@ struct bytes {
  *
  * @return spdlog::logger& The logger.
  */
-RMM_EXPORT inline spdlog::logger& logger()
+[[deprecated(
+  "Support for direct access to spdlog loggers in rmm is planned for "
+  "removal")]] RMM_EXPORT inline spdlog::logger&
+logger()
```

Comment:
Note this makes all the `RMM_LOG_XXX` macros produce warnings, I think.

Reply:
Good point. I can add a level of indirection to help with this.